workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectCreation/SpatialOSEntity.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Improbable.Worker;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public struct SpatialOSEntity
+    {
+        private readonly EntityManager entityManager;
```

Comment:
would prefer to have the `entityManager` not in here. The only use case that I see in this PR for it, has a reference to the worker -> it can easily get the entity manager that way

Reply:
The purpose of this class is to hide ECS stuff from user code that is necessary to write to have GameObjects, if we didn't have the EntityManager we would just not have the class in the first place. and just pass the Entity. But then you'd have to have ECS code necessary to create GameObjects, which is not ideal.

Reply:
However, they still need to know about `IComponentData` to actually use these helper functions. If you want them to use a hybrid workflow you won't get past them using a bit of ECS. In that case, as far as I can see, we want to mainly use this in the `CreateGameObjectForEntity` method. We pass in the worker in this method and therefore have a reference to the world and the entitymanager anyway.

Reply:
The main restriction there is `ISpatialComponentData`, I think it's still useful to keep the exposure to ECS concepts to a minimal level at this point.

Reply:
Tangent: it looks like things may be simpler if we make `ISpatialComponentData` extend `IComponentData` even...